lib/internal/fs/promises.js

```diff
@@ -521,17 +519,22 @@ async function readFileHandle(filehandle, options) {
 
   let size = 0;
   let length = 0;
-  if ((statFields[1/* mode */] & S_IFMT) === S_IFREG) {
-    size = statFields[8/* size */];
+
+  if ((statFields[1 /* mode */] & S_IFMT) === S_IFREG) {
+    size = statFields[8 /* size */];
+
+    if (size > kIoMaxLength) {
+      process.emitWarning(
+        `Warning: Detected \`fs.readFile()\` to read a huge file in memory (${size} bytes). Please consider using \`fs.createReadStream()\` instead to minimize memory overhead and increase the performance.`,
+      );
```

Comment:
It would likely be worthwhile to assign a warning code to this warning so that it can be suppressed with the `--dsable-warning` CLI flag.

![image](https://github.com/user-attachments/assets/ec94215e-e7f7-4729-9c02-69bf1c433d2f)

Reply:
Thanks, ı try to use the ERR_FS_FILE_TOO_LARGE error code, do you think this is true?